kythe/go/platform/kzip/kzip.go

```diff
@@ -95,6 +97,46 @@ import (
 	_ "kythe.io/kythe/proto/java_go_proto"
 )
 
+// Type Encoding describes how compilation units will be encoded when written to a kzip.
+type Encoding int32
+
+const (
+	BOTH  Encoding = 0
+	JSON  Encoding = 1
+	PROTO Encoding = 2
+)
+
+// Set updates an Encoding based on the text value
```

Comment:
Split the flag implementation from the `Encoding` type.  It doesn't really fit in this package.  Add a `EncodingFlag` type to the binary or just handle it directly with a string flag.